src/async_impl/decoder.rs

```diff
@@ -55,12 +70,21 @@ enum Inner {
     #[cfg(feature = "brotli")]
     Brotli(FramedRead<BrotliDecoder<StreamReader<Peekable<IoStream>, Bytes>>, BytesCodec>),
 
+    /// A `Zstd` decoder will uncompress the zstd compressed response content before returning it.
+    #[cfg(feature = "zstd")]
+    Zstd(FramedRead<ZstdDecoder<StreamReader<Peekable<IoStream>, Bytes>>, BytesCodec>),
```

Comment:
I think it makes sense to box the this variant for the same reason in https://github.com/seanmonstar/reqwest/pull/1681